update deprecated MPI functions and fix some warnings

jcosborn opened the pull request:
fix some warnings

detar commented:
Hi James,
Sorry for not noticing your fixes to the MILC code until now.  In the top-level Makefile template, you commented out the LIBQUDA additions relevant to CUDA.  I can see this might be needed for SyCL builds.  Do you not need any CUDA components at all with SyCL builds?   If so, I would instead use the OFFLOAD macro in the Makefile to skip all the definitions involving CUDA.  What do you think?

jcosborn commented:
Yes, SYCL on Intel doesn't need CUDA.  I updated it to only include QUDA CUDA libs based on OFFLOAD (with empty defaulting to CUDA).  I also updated it to reflect the current Khronos convention of SYCL https://www.khronos.org/sycl/ (I don't know why or when it changed).
